src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useRef, useEffect } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types';
+
+interface Props {
+  todo: Todo;
+  isLoading: boolean;
+  isSubmitting: boolean;
+  onDelete: (todoId: number) => void;
+  onUpdate: (todo: Todo) => Promise<void>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isLoading,
+  onDelete,
+  onUpdate,
+}) => {
+  const [isEditingTodo, setIsEditingTodo] = useState<Todo | null>(null);
+  const [newTitle, setNewTitle] = useState('');
+
+  const { title, id, completed } = todo;
+
+  const todoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (todoField.current && isEditingTodo) {
+      todoField.current.focus();
+    }
+  }, [isEditingTodo]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(e.target.value);
+    setIsEditingTodo({ ...isEditingTodo, title: e.target.value } as Todo);
+  };
+
+  const handleTitleUpdate = (newTodo: Todo) => {
+    onUpdate(newTodo).then(() => setIsEditingTodo(null));
+  };
+
+  const handleEvent = (eventKey?: string) => {
+    const event = eventKey === undefined || eventKey === 'Enter';
+
+    if (event && isEditingTodo && isEditingTodo.title.length) {
+      if (isEditingTodo.title.trim() !== todo.title) {
+        handleTitleUpdate({ ...isEditingTodo, title: newTitle.trim() });
+      } else {
+        setIsEditingTodo(null);
+      }
+    }
+
+    if (event && isEditingTodo?.title.length === 0) {
```

Comment:
```suggestion
    if (event && !isEditingTodo?.title.length) {
```

Reply:
I don't now why, but title not changed when i write `!isEditingTodo`, its work only with `isEditingTodo?.title.length === 0`